derive/src/impl_wrapper.rs

```diff
@@ -12,32 +12,36 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#[cfg(not(feature = "std"))]
-use alloc::{
-    format,
-    string::ToString,
-};
-
 use proc_macro2::{
     Span,
     TokenStream as TokenStream2,
 };
 use quote::quote;
 use syn::Ident;
 
-pub fn wrap(
-    ident: &Ident,
-    trait_name: &'static str,
-    impl_quote: TokenStream2,
-) -> TokenStream2 {
-    let mut renamed = format!("_IMPL_{}_FOR_", trait_name);
-    renamed.push_str(ident.to_string().trim_start_matches("r#"));
-    let dummy_const = Ident::new(&renamed, Span::call_site());
+pub fn wrap(impl_quote: TokenStream2) -> TokenStream2 {
+    let include_scale_info = include_crate("scale-info", "_scale_info");
+    let include_parity_scale_codec = include_crate("parity-scale-codec", "_scale");
```

Comment:
This is in general a bad style, to require some other crate than the macro providing crate in the `Cargo.toml`.

Why aren't you re-exporting the required stuff from `scale-info`?

Reply:
I am not entirely sold on this.
While I agree that your proposal solves certain problems the `scale-info` crate works with multiple `parity-scale-codec` versions. So re-exporting one of the supported versions would case a lock-in and potentially a duplication of dependencies in case the `scale-info` user does not also use the same version.

The reality is that `scale-info` and `parity-scale-codec` are somehow tightly coupled by design (one does encoding, the other does encoding of the encoding) and could actually even live under the same crate.

Reply:
Indeed that is a much better idea